src/api/core/warren/xi/factories.py

```diff
@@ -0,0 +1,170 @@
+
+from warren.fields import IRI
+from warren.xi.models import (
+    Experience,
+    Relation,
+)
+
+T = TypeVar("T")
+
+
+class SyncPersistenceHandler(SyncPersistenceProtocol[T]):
+    """Sync persistence handler for SQLModelFactory."""
+
+    def __init__(self, session: Session) -> None:
+        """Initializes SyncPersistenceHandler with a SQLAlchemy Session.
+
+        Args:
+            session (Session): The SQLAlchemy session to be used for db operations.
+        """
+        self.session = session
+
+    def save(self, data: T) -> T:
+        """Saves a single instance of data to the database.
+
+        Args:
+            data (T): The data instance to be saved.
+
+        Raises:
+            AttributeError: If the session is not set.
+
+        Returns:
+            T: The saved data instance.
+        """
+        if self.session is None:
+            raise AttributeError("Session is not set")
+
+        self.session.add(data)
+        self.session.commit()
+        return data
+
+    def save_many(self, data: List[T]) -> List[T]:
+        """Saves multiple instances of data to the database.
+
+        Args:
+            data (List[T]): The list of data instances to be saved.
+
+        Raises:
+            AttributeError: If the session is not set.
+
+        Returns:
+            List[T]: The list of saved data instances.
+        """
+        if self.session is None:
+            raise AttributeError("Session is not set")
+        for d in data:
+            self.session.add(d)
+        self.session.commit()
+        return data
+
+
+class SQLModelFactory(Generic[T], ModelFactory[T]):
+    """Base factory for SQLModel models."""
+
+    __sync_persistence__ = SyncPersistenceHandler[T]
+    __session__: Session = None
+
+    __is_base_factory__ = True
+
+    @classmethod
+    def _get_sync_persistence(cls) -> SyncPersistenceProtocol[T]:
+        """Initialize the SyncPersistenceHandler with the session."""
+        return SyncPersistenceHandler(cls.__session__)
+
+
+class IgnoreAutogeneratedFieldsMixin:
+    """Mixin class to ignore autogenerated fields by SQLAlchemy.
+
+    This class can be used as a mixin to exclude specific fields
+    (such as id, created_at, updated_at) from being generated by factory.
+    """
+
+    id = Ignore()
+    created_at = Ignore()
+    updated_at = Ignore()
+
+
+class ExperienceFactory(SQLModelFactory[Experience], IgnoreAutogeneratedFieldsMixin):
+    """Factory for Experience model."""
+
+    __model__ = Experience
+
+    language = Use(SQLModelFactory.__random__.choice, ["fr", "en-GB", "en"])
+
+    @classmethod
+    def get_provider_map(cls) -> Dict[Type, Any]:
+        """Extends the provider map from the Base class with the IRI type."""
+        providers_map = super().get_provider_map()
+        return {
+            IRI: lambda: f"uuid://{uuid4().hex}",
+            **providers_map,
+        }
+
+    @classmethod
+    def build_dict(cls, exclude=None, json_dump=None, **kwargs) -> dict:
```

Comment:
What about using Pydantic-style API?

```suggestion
    def dict(cls, exclude=None, json_dump=None, **kwargs) -> dict:
```

Reply:
At the beginning, I thought about writing it that way. I am totally open to rename the method, no strong opinion on my side. 
To justify this choice. I feel this naming give a more accurate description of what is happening in the class method. Plus, I would find confusing to reuse Pydantic method's naming on a non-pydantic instance. Wdyt?

Reply:
Yes, you are right. Let's leave it that way.